internal/osbuild2/sshd_config_stage.go

```diff
@@ -1,9 +1,82 @@
 package osbuild2
 
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
 type SshdConfigConfig struct {
-	PasswordAuthentication          *bool `json:"PasswordAuthentication,omitempty"`
-	ChallengeResponseAuthentication *bool `json:"ChallengeResponseAuthentication,omitempty"`
-	ClientAliveInterval             *int  `json:"ClientAliveInterval,omitempty"`
+	PasswordAuthentication          *bool                `json:"PasswordAuthentication,omitempty"`
+	ChallengeResponseAuthentication *bool                `json:"ChallengeResponseAuthentication,omitempty"`
+	ClientAliveInterval             *int                 `json:"ClientAliveInterval,omitempty"`
+	PermitRootLogin                 PermitRootLoginValue `json:"PermitRootLogin,omitempty"`
+}
+
+// PermitRootLoginValue is defined to represent all valid types of the
+// 'PermitRootLogin' item in the SshdConfigConfig structure.
+type PermitRootLoginValue interface {
+	isPermitRootLoginValue()
+}
+
+// PermitRootLoginValueStr represents a string type of the 'PermitRootLogin'
+// item in the SshdConfigConfig structure.
+type PermitRootLoginValueStr string
+
+func (v PermitRootLoginValueStr) isPermitRootLoginValue() {}
+
+// PermitRootLoginValueBool represents a bool type of the 'PermitRootLogin'
+// item in the SshdConfigConfig structure.
+type PermitRootLoginValueBool bool
+
+func (v PermitRootLoginValueBool) isPermitRootLoginValue() {}
+
+// Valid values which can be used for the 'PermitRootLogin' item in
+// the SshdConfigConfig structure.
+const (
+	PermitRootLoginValueYes PermitRootLoginValueBool = true
```

Comment:
A very valid question. This was in fact discussed in the related osbuild PR and the reason for using bool was to keep consistency with other options in the stage which do not accept string values and use only bool for "yes" / "no". More info is in https://github.com/osbuild/osbuild/pull/917

Reply:
We have done this before where the stage in osbuild supports more formats or types than the one in composer.  I think in this case we decided it would be strange if booleans weren't supported at all.